modules/videoio/test/test_ffmpeg.cpp

```diff
@@ -636,5 +636,54 @@ const ffmpeg_cap_properties_param_t videoio_ffmpeg_properties[] = {
 
 INSTANTIATE_TEST_CASE_P(videoio, ffmpeg_cap_properties, testing::ValuesIn(videoio_ffmpeg_properties));
 
+// related issue: https://github.com/opencv/opencv/issues/15499
+TEST(videoio, mp4_orientation_meta_auto)
+{
+    string video_file = string(cvtest::TS::ptr()->get_data_path()) + "video/big_buck_bunny_rotated.mp4";
+
+    VideoCapture cap;
+    EXPECT_NO_THROW(cap.open(video_file, CAP_FFMPEG));
+    ASSERT_TRUE(cap.isOpened()) << "Can't open the video: " << video_file << " with backend " << CAP_FFMPEG << std::endl;
+    ASSERT_TRUE(cap.get(CAP_PROP_ORIENTATION_AUTO));
+
+    Size actual;
```

Comment:
I suppose to check `ASSERT_TRUE(cap.get(CAP_PROP_ORIENTATION_META) == 90);`
before doing next steps